core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1876,6 +1920,22 @@ abstract class RDD[T: ClassTag](
  */
 object RDD {
 
+  /**
+   * The random level of RDD's computing function, which indicates the behavior when rerun the
+   * computing function. There are 3 random levels, ordered by the randomness from low to high:
+   * 1. IDEMPOTENT: The computing function always return the same result with same order when rerun.
+   * 2. UNORDERED: The computing function returns same data set in potentially a different order
+   *               when rerun.
+   * 3. INDETERMINATE. The computing function may return totally different result when rerun.
+   *
+   * Note that, the output of the computing function usually relies on parent RDDs. When a
+   * parent RDD's computing function is random, it's very likely this computing function is also
+   * random.
+   */
+  object RandomLevel extends Enumeration {
```

Comment:
While reviewing the PR, this enumeration looks a bit unclear.
We are modelling two things here :
* Behavior of `compute` in RDD - whether it is idempotent, order sensitive or indeterminate.
also,
* What is the input order of the tuples coming in from the partition to `RDD.compute`

For example, shuffle output of a sorted RDD would be `IDEMPOTENT` by this definition - but idempotent is a functional behavior of a closure, not input order. Perhaps we need to convey this better ? Or does 'idempotent' order make sense ?
Thoughts @cloud-fan, @markhamstra ?